### PR TITLE
chore(ci): Add docker check  

### LINKED ISSUE #131: Add tests for docker build
## Description
Whenever changes occur to any of the docker files, we want a CI to run to evaluate the validity of these changes. Right now, we have just 2 docker files:
- `apps/web/Dockerfile`
- `apps/api/Dockerfile`

## Solution

- The CI should run when: 
  1. Changes occur to any of the docker file
  2. Changes are made during a push or in a pull request
- Test for build: Issue a `docker build` command, then check if the image was built successfully

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
## Description
CI which checks docker build whenever changes are done in docker file 

Fixes #131

## Dependencies
N/A

## Future Improvements
N/A

## Mentions
@rajdip-b 

## Screenshots of relevant screens

Screenshot from 2024-07-27 17-47-29

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

File: apps/web/Dockerfile [hunk 1]
```diff
@@ -28,6 +28,7 @@ COPY --from=builder /app/out/full/ .
 COPY turbo.json turbo.json
 
 
+
```
Comment (DhairyaMajmudar) [inline apps/web/Dockerfile:31]: change kept to check the ci, will remove this once PR is approved

File: .github/workflows/docker-ci.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+on:
+  push:
+    paths:
+      - '**Dockerfile'
+  pull_request:
+    paths:
+      - '**Dockerfile'
+jobs:
+  build-image-1:
```
Comment (rajdip-b) [inline .github/workflows/docker-ci.yml:11]: Change the job name to `build`

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-ci.yml (hunk 2 above)
Comment (DhairyaMajmudar) [reply .github/workflows/docker-ci.yml]: Done 🚀

**Review by rajdip-b — APPROVED** [review]
LGTM! Will merge it in the review window.

**Pull request merged**